### ISSUE TITLE
StackOverflow exception thrown when invoking StartCopyAsync on a PageBlob

### ISSUE THREAD

**Issue by Jericho** [Azure/azure-storage-net#597]
Whenever I try to copy a page blob, I get a StackOverflow exception and my console app crashes. This issue can be easily reproduced with the following `program.cs`:

```csharp
using Microsoft.WindowsAzure.Storage;
using System;
using System.Threading;

namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");

            var storageAccount = CloudStorageAccount.DevelopmentStorageAccount;
            var blobClient = storageAccount.CreateCloudBlobClient();
            var blobContainer = blobClient.GetContainerReference("mycontainer");

            var sourceBlob = blobContainer.GetPageBlobReference("SOURCE.txt");
            var targetBlob = blobContainer.GetPageBlobReference("TARGET.txt");
            targetBlob.StartCopyAsync(sourceBlob, null, null, null, null, CancellationToken.None).Wait();

        }
    }
}
```

and with the following `ConsoleApp1.csproj`:
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="WindowsAzure.Storage" Version="8.7.0" />
  </ItemGroup>

</Project>

```

Here's the result when I run this console app: 
![sshot-12](https://user-images.githubusercontent.com/112710/34364117-86e6887c-ea50-11e7-9743-36a4f89b6b97.png)

The output window in Visual Studio shows the following:

> An unhandled exception of type 'System.StackOverflowException' occurred in Microsoft.WindowsAzure.Storage.dll

**Comment by erezvani1529** [comment]
Hi @Jericho,

Thank you very much for reporting this issue. Looks like the overload that you are using has regressed since a few releases back. We will release the fix for it asap. 

As a workaround, please use the maximum [overload](https://github.com/Azure/azure-storage-net/blob/master/Lib/WindowsRuntime/Blob/CloudPageBlob.cs#L1230) for StartCopyAsync:

`targetBlob.StartCopyAsync(sourceBlob, premiumBlobTier: null, sourceAccessCondition: null, destAccessCondition: null, options: null, operationContext: null, cancellationToken: CancellationToken.None);`

**Comment by Jericho** [comment]
@erezvani1529 I just want to confirm that the workaround you suggested solves the issue. Thank you.

**Comment by erezvani1529** [comment]
@Jericho ,

This issue is fixed with [v9.0.0.](https://www.nuget.org/packages/WindowsAzure.Storage/9.0.0) Thanks again for reporting.
